chapter3/parsec.hs

```diff
@@ -54,9 +54,9 @@ failure = Parser (\cs -> [])
 
 option :: Parser a -> Parser a -> Parser a
 option  p q = Parser $ \s ->
-  case parse (mplus p q) s of
-    []     -> []
-    (x:xs) -> [x]
+  case parse p s of
```

Comment:
``mzero`` wasn't a neutral element for ``(<|>)`` - see message of commit 1170925.
It makes no difference as long as all our parsers return at most one parse.